### ISSUE TITLE
Create Sdist

### ISSUE THREAD

**Issue by gettoknowmii** [czue/celery-progress#85]
At https://pypi.org/project/celery-progress/#files there is no sdist, as is standard pypi/python practice.

Any plans to create one? It would greatly simplify the work of your downstream packagers.

Thank you.

**Comment by czue** [comment]
Oops sorry about that. Definitely happy to do this for the next release. For my own education, what does the sdist help with that isn't covered by the wheel?

**Comment by dlangille** [comment]
I am a FreeBSD ports committer but I do not have great insight into Python packages. I know the FreeBSD packaging system (known as ports) uses the sdist file.  To get more information than that for you, I've put the question out to other FreeBSD port developers.

**Comment by neirbowj** [comment]
I'll take a shot here.

The wheel format is itself designed to be an installable package. The FreeBSD ports machinery is designed to produce an installable package of its own format. In order to use a wheel as the input to the FreeBSD port build process, the build machinery would have to be able to disassemble one package format in order to be able to re-assemble the contents into another package format.

On top of that, wheels in general can contain platform- and architecture-specific files that are produced during the build process. Consequently, the Python parts of the FreeBSD ports machinery would need to be able to select the right wheel for a given build context (Python version, options, architecture, etc), and port maintainers would still need to be able to fall back to configuring their ports to build from sdists in cases when the upstream project does not publish a usable wheel.

In summary, sdist is both better suited as the input to a build process that produces a package than a wheel is, and is much more reliable common denominator

For reference, Python ports make up roughly 11% of the ~29k ports in the FreeBSD ports tree.

**Comment by koobs** [comment]
* Allows consumers (both downstream packagers, and users) to customise the build, appropriate for their needs or environment
* Ensures upstreams are testing a complete Python packaging / distribution pipeline (compliance, best practice, etc)

See Also:

 * https://discuss.python.org/t/purpose-of-an-sdist/4639

**Comment by czue** [comment]
:+1: thanks for the responses! I'll try and make this happen this coming week. :crossed_fingers:

**Comment by czue** [comment]
I think this is done now (I also added a new release to be sure). Can you check and close if it looks ok?

**Comment by czue** [comment]
closing this - though let me know if you need anything else

**Comment by dlangille** [comment]
@czue :  Confirmed works.

I committed a FreeBSD port yesterday: https://www.freshports.org/devel/py-celery-progress/

Thank you for your help. It is appreciated.

**Comment by czue** [comment]
great!
